pipelines/data_pipelines.go

```diff
@@ -0,0 +1,43 @@
+package pipelines
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/pipelines"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourcePipelines() *schema.Resource {
+	type pipelineData struct {
+		Name string            `json:"pipeline_name,omitempty"`
+		Ids  map[string]string `json:"ids,omitempty" tf:"computed"`
+	}
+	return common.WorkspaceData(func(ctx context.Context, data *pipelineData, w *databricks.WorkspaceClient) error {
+		errorMessage := fmt.Errorf("no pipelines found")
+		pipelineSearch := pipelines.ListPipelines{MaxResults: 100}
+
+		if data.Name != "" {
+			searchPattern := fmt.Sprintf("name LIKE '%s'", data.Name)
+			errorMessage = fmt.Errorf("there is no pipeline with name LIKE '%s'; you need to specify `pipeline_name` as the full pipeline name or with percent wildcards", data.Name)
+			pipelineSearch = pipelines.ListPipelines{Filter: searchPattern, MaxResults: 100}
+		}
+
+		pipelines, err := w.Pipelines.ListPipelinesAll(ctx, pipelineSearch)
```

Comment:
please use https://pkg.go.dev/github.com/databricks/databricks-sdk-go@v0.7.0/service/pipelines#PipelinesAPI.PipelineStateInfoNameToPipelineIdMap, which also performs name de-duplication.

Reply:
> please use https://pkg.go.dev/github.com/databricks/databricks-sdk-go@v0.7.0/service/pipelines#PipelinesAPI.PipelineStateInfoNameToPipelineIdMap, which also performs name de-duplication.

#nfx Pipeline names are not unique. See "test" pipelines in screenshot. In this scenario, should we throw an error when searching by pipeline name "test" or return all pipelines that match that search criteria?
<img width="968" alt="Screenshot 2023-04-12 at 10 13 45 AM" src="https://user-images.githubusercontent.com/8633074/231486638-2d7a287a-fd1c-44e5-b3d0-6b7b1d8ca59b.png">

Reply:
imho, for `databricks_pipelines` we should return all of them.

Reply:
that's my thought as well. If that's the case then we should return id-to-name map (ID as key) from this data provider to allow for duplicate names.